### ISSUE TITLE
`.schema` support

### ISSUE THREAD

**Issue by vesse** [airtoxin/purify-ts-extra-codec#8]
`purify-ts` can generate valid JSON schemas from codecs, and JSON schema supports many of the options these extra codecs have defined. It would be cool if these codecs had the `.schema` method too and could produce eg.

```
{ 
  type: 'string',
  maxLength: 10
}
```

from `StringLengthRangedIn({ lte: 10 })`

**Comment by airtoxin** [comment]
nice.
I promise making its changes but my computer was broken now, so it takes some days.
or pull request welcome

**Comment by vesse** [comment]
I was thinking about pull request but didn't quickly figure out how schema would work with chaining codecs and decide if should change `extendCodec` signature or wrap just the built in codecs with another wrapper that adds the schema.

**Comment by airtoxin** [comment]
@vesse I'm checking purify-ts documents to set JSON schema, but there are no way to set specific schema from out of purify-ts, so this feature can't realize. sorry 😢

**Comment by vesse** [comment]
Hmm, what do you mean there is no way to set schema? I needed `StringLengthRangedIn` with schema support but with less options so I did this:

```typescript
const rangedLengthString = ({ gte, lte }: RangedLengthStringOptions): Codec<string> =>
  Codec.custom<string>({
    decode: (value) =>
      string.decode(value).chain((value) => {
        const length = value.length
        if (gte != null && length < gte) {
          return Left(`length of ${value} must be greater than equal ${gte}`)
        }
        if (lte != null && length > lte) {
          return Left(`length of ${value} must be less than equal ${lte}`)
        }

        return Right(value)
      }),
    encode: string.encode,
    schema: () => ({
      type: 'string',
      ...(gte ? { minLength: gte } : undefined),
      ...(lte ? { maxLength: lte } : undefined),
    }),
  })
```

which as you can see is pretty similar to yours but with added `schema` function which produces the JSON schema I needed for my Fastify route handler.

**Comment by airtoxin** [comment]
ahh I see. schema option is undocumented so I missed.

**Comment by airtoxin** [comment]
@vesse 
I added basic functionality support of schema.
Released v0.6.0 https://github.com/airtoxin/purify-ts-extra-codec/releases

**Comment by vesse** [comment]
Awesome, thanks!
